From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Antonio=20V=C3=A1zquez=20Blanco?=
 <antoniovazquezblanco@gmail.com>
Date: Mon, 20 Jul 2026 10:48:26 +0200
Subject: [PATCH] 9398: Bug: Fix radix mismatch in GenericAddress sub-word
 offset display

Fix radix mismatch in GenericAddress sub-word offset display
---
 .../java/ghidra/program/model/address/GenericAddress.java     | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/address/GenericAddress.java b/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/address/GenericAddress.java
index f2de13f3cd..983d5a5d83 100644
--- a/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/address/GenericAddress.java
+++ b/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/address/GenericAddress.java
@@ -281,7 +281,7 @@ public class GenericAddress implements Address {
 		}
 		long mod = 0;
 		if (unitSize > 1) {
-			mod = displayOffset % unitSize;
+			mod = Long.remainderUnsigned(displayOffset, unitSize);
 			displayOffset = addrSpace.getAddressableWordOffset(displayOffset);
 		}
 
@@ -294,7 +294,7 @@ public class GenericAddress implements Address {
 		buf.append(addressString);
 		if (mod != 0) {
 			buf.append('.');
-			buf.append(mod);
+			buf.append(Long.toHexString(mod));
 		}
 		if (stackFormat) {
 			buf.append(STACK_ADDRESS_SUFFIX);
-- 
2.45.1

